WordPress/gutenberg
# ToggleControl: Convert to TypeScript

Part of #35744 #38730

## What?

- Migrate ToggleControl (component, stories, tests) to TypeScript.
- Remove custom CSS in favor of HStack.
- Add a `__nextHasNoMarginBottom` prop for removing margins.

## Why?

Better devex.

## Testing Instructions

`npm run storybook:dev` and check the stories for ToggleControl.

Note that the space between the toggle and the `help` text is narrower than on trunk. This is intentional, to address https://github.com/WordPress/gutenberg/pull/39209#issuecomment-1072175095.
